scripts/process_new_format.py

```
@@ -171,65 +180,223 @@ def process_markdown_faq_data(markdown_faq_data: dict[str, str]) -> None:
+    """
+    Processes a BeautifulSoup object representing a single ruling item (e.g., content of an <li>)
+    and extracts structured Ruling objects.
+    """
+    rulings_for_item = []
+    current_question_html_str = None # Store HTML string of the question part
+
+    original_snippet_html = ruling_soup.decode_contents().strip()
+
+    # Basic Provenance for ArkhamDB
+    base_provenance = Provenance(
+        source_type="arkhamdb_faq",
```

Ruff (0.11.9)

Line too long (101 > 100)

(E501)